ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/StreamInfoImpl.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.StreamInfo;
+import java.util.Objects;
+
+public class StreamInfoImpl implements StreamInfo {
```

Comment:
generally worth having equals/hashcode/tostring on externally exposed classes so that people can leverage them if they want to

Reply:
To make sure I understand correctly: you're suggesting we add equals/hashcode/tostring methods to each of the implementation classes, not the interfaces, right? This makes sense to me, though I'm not sure whether it makes more sense for the toString method to say `StreamInfoImpl{...}` or `StreamInfo{...}` since the former is an implementation detail, but the latter is a bit misleading.

Reply:
Hmm, good point - I'm satisfied with `StreamInfo` because I dont' think we expect more than just this impl

Reply:
Sounds good. I'm going to merge this PR and then open a follow-up to add equals/hashcode/tostring to all the client interfaces, since none of the new interfaces have them (not just the ones in this PR).

Reply:
Here's the follow-up: https://github.com/confluentinc/ksql/pull/5681